autoload/lsp/lspserver.vim

```diff
@@ -862,8 +862,12 @@ enddef
 
 # process the 'textDocument/documentHighlight' reply from the LSP server
 # Result: DocumentHighlight[] | null
-def DocHighlightReply(bnr: number, lspserver: dict<any>, docHighlightReply: any): void
+def DocHighlightReply(lspserver: dict<any>, docHighlightReply: any,
+                      bnr: number, cmdmods: string): void
   if docHighlightReply->empty()
+    if cmdmods !~ 'silent'
+      util.WarnMsg($'No highlight for the current position')
```

Comment:
When the "autoHighlight" option is set to true, this message will be displayed if the current position doesn't have a highlight.  So this message will be displayed many times.

Reply:
Makes sense, I have changed the code to send `silent` to the `LspDocHighlight` function from within the CursorMoved autocmd.